Extension/src/LanguageServer/client.ts

```diff
@@ -2364,19 +2364,23 @@ export class DefaultClient implements Client {
         this.notifyWhenReady(() => this.languageClient.sendNotification(ChangeCompileCommandsNotification, params));
     }
 
-    private isSourceFileConfigurationItem(input: any): input is SourceFileConfigurationItem {
+    private isSourceFileConfigurationItem(input: any, providerVersion: Version): input is SourceFileConfigurationItem {
+        const intelliSenseModeValid = providerVersion > Version.v4 ?
+            util.isOptionalString(input.configuration.intelliSenseMode) : util.isString(input.configuration.intelliSenseMode)
```

Comment:
It would probably be sufficient to use isOptionalString in both cases.  The prior API did not allow undefined, but it wouldn't hurt to handle it anyway.